maroon/src/linearizer.rs

```diff
@@ -0,0 +1,101 @@
+use crate::epoch::Epoch;
+use common::range_key::UniqueU64BlobId;
+use log::debug;
+
+pub trait Linearizer {
+  fn new_epoch(&mut self, epoch: Epoch);
+}
+
+pub struct LogLineriazer {
+  sequence: Vec<UniqueU64BlobId>,
+}
+
+impl LogLineriazer {
+  pub fn new() -> LogLineriazer {
+    LogLineriazer { sequence: vec![] }
+  }
+}
+
+impl Linearizer for LogLineriazer {
+  fn new_epoch(&mut self, mut epoch: Epoch) {
+    epoch.increments.sort();
```

Comment:
This is confusing.

We append intervals so that the next one is right after the previous one, right? As in, it will always be a single `[left, right]` (or `[begin, end)`) range, since if it is not there are some serious issues with ordering.

Or what am I missing?

Reply:
No. Becausethe order is not guaranteed here it might be:
[
(0,10),
(4523, 4570),
(8141,8241)
]

but it can also be

[
(8141,8241),
(0,10),
(4523, 4570)
]

Or any other order. That happens because when I create this slice I iterate over HashMap which doesn't guarantee keys order.
I can actually enforce the order(for example creating a special type that will keep the order sorted all the time). That's an option